plugins/woocommerce/tests/php/src/Blocks/Shipping/ShippingControllerTest.php

```diff
@@ -90,4 +128,52 @@ function ( $locale ) {
 		// Remove filter.
 		remove_all_filters( 'woocommerce_get_country_locale' );
 	}
+
+	/**
+	 * Test that register_local_pickup handles missing WC()->shipping and other dependencies gracefully.
+	 */
+	public function test_register_local_pickup_also_handles_missing_dependencies() {
+		$wc_backup   = WC();
+		$logger_mock = $this->getMockBuilder( \WC_Logger_Interface::class )->getMock();
+		// We need this to make sure our mock logger is getting returned.
+		add_filter(
+			'woocommerce_logging_class',
+			function () use ( $logger_mock ) {
+				return $logger_mock;
+			},
+			10,
+			0
+		);
+
+		try {
+			// Test that the method does not throw exceptions without missing dependencies.
+			$this->shipping_controller->register_local_pickup();
+			$this->assertTrue( true, 'Method did not throw exceptions without missing dependencies' );
```

Comment:
Why are we doing this? I'm not sure these particular logs add any value, if the test fails, we would see an error indicating why, and the stack trace would point us to which call the error originated from.

Reply:
The assertion implies the intention behind the test, that we are specifically testing it to not throw.